pkg/storage/elasticsearch.go

```diff
@@ -122,11 +123,30 @@ func (ed *ElasticsearchDeployment) InjectStorageConfiguration(p *corev1.PodSpec)
 
 // InjectSecretsConfiguration changes the given spec to include the options for the index cleaner
 func (ed *ElasticsearchDeployment) InjectSecretsConfiguration(p *corev1.PodSpec) {
+	secretName := curatorSecret.instanceName(ed.Jaeger)
+
+	// NOTE: If the permissions are more finely granulated, the writing of the
+	// dependency store must be skipped, since these are not currently granted
+	// to the sg_jaeger role.
+	// This implies an jaeger version higher than 1.35.1.
+	// Since this will not be shipped with the next operator version, this
```

Comment:
add TODO with versions when it was added and when it can be removed - if we know and can be sure ES will be released with the fix we need for Jaeger.